src/python-common/ceph/deployment/service_spec.py

```diff
@@ -567,6 +567,12 @@ def service_name(self) -> str:
             n += '.' + self.service_id
         return n
 
+    def get_ports(self) -> List[int]:
+        return []
```

Comment:
The ports are per-daemon, but `ServiceSpec` is per Service. If you add this **only** to this class, it's going to be painful to support different ports for different daemons. 

Also, we have default ports for different service types. and the MGR is super special.